src/gateway/handlers/guildMemberAdd.ts

```diff
@@ -11,6 +12,6 @@ export const guildMemberAdd: GatewayEventHandler = async (
   if (guild === undefined) return
 
   await guild.members.set(d.user.id, d)
-  const member = await guild.members.get(d.user.id)
-  gateway.client.emit('guildMemberAdd', member)
+  const member = await guild.members.get(d.user.id) as unknown
```

Comment:
Why did you set this as unknown? If you did this because of undefined you should handle it instead of `as unknown`

Reply:
Can you read like 14? We're clearly setting the user to cache, then getting it from cache to it returns a proper member object. No chance for it to be undefined.

Reply:
Why cache when you can directly fetch from gateway 👀

Reply:
Than it can be just Member instead of unknown isn't it?

Reply:
Yep, member =! unknown

Reply:
Oh, yeah. I'll make it `as Member`. I thought linter going to yell at me for unnecessary type assertion 🤔 but different case 😂

Reply:
> Why cache when you can directly fetch from gateway 👀

Caching is good practice to avoid making too much requests to API/Gateway.

Reply:
> > Why cache when you can directly fetch from gateway 👀
> 
> Caching is good practice to avoid making too much requests to API/Gateway.

Just don’t make it fully like discord.js, or else this lib will be shitty.

Reply:
> Just don’t make it fully like discord.js, or else this lib will be shitty.

In my opinion that makes no sense. Almost every lib does caching, but most provide a way to control. You can do it with intents, omit the intent, event won't be fired, caching won't be done, simple. This lib even provides a way to setup custom caching, you can also disable certain caches using that too - simply return and not save cache if the cacheName is in disabled ones. But personally - caching is not a bad thing, just that discord.js doesn't provide much control, I mean it's still a good lib.

Reply:
I only make bots to play with the gateway, so I don’t really need caching but thanks for implementing no caching methods.